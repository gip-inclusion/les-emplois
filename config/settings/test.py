import os


ITOU_ENVIRONMENT = "DEV"
os.environ["ITOU_ENVIRONMENT"] = ITOU_ENVIRONMENT

from .base import *  # noqa: E402,F403


SECRET_KEY = "foobar"
ALLOWED_HOSTS = []

# We *want* to do the same `collectstatic` on the CI than on PROD to catch errors early,
# but we don't want to do it when running the test suite locally for performance reasons.
if not os.getenv("CI", False):
    # `ManifestStaticFilesStorage` (used in base settings) requires `collectstatic` to be run.
    STORAGES["staticfiles"]["BACKEND"] = "django.contrib.staticfiles.storage.StaticFilesStorage"  # noqa: F405

ASP_ITOU_PREFIX = "XXXXX"  # same as in our fixtures
ITOU_PROTOCOL = "http"
ITOU_FQDN = "localhost:8000"

DATABASES["default"]["HOST"] = os.getenv("PGHOST", "127.0.0.1")  # noqa: F405
DATABASES["default"]["PORT"] = os.getenv("PGPORT", "5432")  # noqa: F405
DATABASES["default"]["NAME"] = os.getenv("PGDATABASE", "itou")  # noqa: F405
DATABASES["default"]["USER"] = os.getenv("PGUSER", "postgres")  # noqa: F405
DATABASES["default"]["PASSWORD"] = os.getenv("PGPASSWORD", "password")  # noqa: F405

MAILJET_API_KEY_PRINCIPAL = "API_MAILJET_KEY_PRINCIPAL"
MAILJET_SECRET_KEY_PRINCIPAL = "API_MAILJET_SECRET_PRINCIPAL"

AWS_S3_ENDPOINT_URL = f"http://{os.getenv('CELLAR_ADDON_HOST', 'localhost:9000')}/"
AWS_S3_ACCESS_KEY_ID = "minioadmin"
AWS_S3_SECRET_ACCESS_KEY = "minioadmin"
AWS_STORAGE_BUCKET_NAME = "tests"
