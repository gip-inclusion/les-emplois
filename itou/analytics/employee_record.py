from django.db.models import Count, Max, Min

from itou.analytics import models
from itou.employee_record import models as employee_record_models


def collect_employee_records_count(before):
    return {
        models.DatumCode.EMPLOYEE_RECORD_COUNT: employee_record_models.EmployeeRecord.objects.filter(
            created_at__lt=before
        ).count(),
    }


def collect_probably_deleted_employee_records(before):
    data = employee_record_models.EmployeeRecord.objects.filter(created_at__lt=before).aggregate(
        first_pk=Min("pk"),
        last_pk=Max("pk"),
        count_pk=Count("pk"),
    )

    value = ((data["last_pk"] - data["first_pk"] + 1) - data["count_pk"]) if None not in data.values() else 0
    return {models.DatumCode.EMPLOYEE_RECORD_DELETED: value}


def collect_employee_records_processing_code_of_first_exchange(before):
    data = (
        employee_record_models.EmployeeRecord.objects.filter(created_at__lt=before)
        .values("asp_processing_code")
        .annotate(
            count=Count("asp_processing_code"),
        )
    )
    count_by_processing_code = {item["asp_processing_code"]: item["count"] for item in data}

    return {
        models.DatumCode.EMPLOYEE_RECORD_PROCESSED_AT_FIRST_EXCHANGE: count_by_processing_code.get(
            employee_record_models.EmployeeRecord.ASP_PROCESSING_SUCCESS_CODE, 0
        ),
        models.DatumCode.EMPLOYEE_RECORD_WITH_ERROR_AT_FIRST_EXCHANGE: sum(
            count
            for code, count in count_by_processing_code.items()
            if code != employee_record_models.EmployeeRecord.ASP_PROCESSING_SUCCESS_CODE
        ),
        models.DatumCode.EMPLOYEE_RECORD_WITH_ERROR_3436_AT_FIRST_EXCHANGE: count_by_processing_code.get(
            employee_record_models.EmployeeRecord.ASP_DUPLICATE_ERROR_CODE, 0
        ),
    }


def collect_employee_record_with_at_least_one_error(before):
    employee_records_with_at_least_one_error = employee_record_models.EmployeeRecordQuerySet.union(
        employee_record_models.EmployeeRecord.objects.filter(created_at__lt=before)
        .exclude(
            asp_processing_code=employee_record_models.EmployeeRecord.ASP_PROCESSING_SUCCESS_CODE,
        )
        .values("pk"),
        employee_record_models.EmployeeRecordUpdateNotification.objects.filter(created_at__lt=before)
        .exclude(
            asp_processing_code=employee_record_models.EmployeeRecord.ASP_PROCESSING_SUCCESS_CODE,
        )
        .values("employee_record"),
    )

    value = (
        employee_record_models.EmployeeRecord.objects.filter(pk__in=employee_records_with_at_least_one_error)
        .distinct()
        .count()
    )
    return {models.DatumCode.EMPLOYEE_RECORD_WITH_AT_LEAST_ONE_ERROR: value}


def collect_analytics_data(before):
    return {
        **collect_employee_records_count(before),
        **collect_probably_deleted_employee_records(before),
        **collect_employee_records_processing_code_of_first_exchange(before),
        **collect_employee_record_with_at_least_one_error(before),
    }
