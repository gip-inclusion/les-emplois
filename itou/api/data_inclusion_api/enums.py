import enum


class StructureTypeStr(str, enum.Enum):
    ORGA = "orga"
    SIAE = "siae"
