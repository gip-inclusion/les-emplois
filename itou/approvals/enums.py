from django.db import models

from itou.companies import enums as companies_enums


class ApprovalStatus(models.TextChoices):
    EXPIRED = "EXPIRED", "Expiré"
    VALID = "VALID", "Valide"
    FUTURE = "FUTURE", "Valide (non démarré)"
    SUSPENDED = "SUSPENDED", "Valide (suspendu)"


class Origin(models.TextChoices):
    DEFAULT = "default", "Créé normalement via les emplois"
    PE_APPROVAL = "pe_approval", "Créé lors d'un import d'Agrément Pole Emploi"
    # On November 30th, 2021, AI were delivered approvals without a diagnosis.
    AI_STOCK = "ai_stock", "Créé lors de l'import du stock AI"
    ADMIN = "admin", "Créé depuis l'admin"


class ProlongationReason(models.TextChoices):
    SENIOR_CDI = "SENIOR_CDI", "CDI conclu avec une personne de plus de 57 ans"
    COMPLETE_TRAINING = "COMPLETE_TRAINING", "Fin d'une formation"
    RQTH = "RQTH", "RQTH - Reconnaissance de la qualité de travailleur handicapé"
    SENIOR = "SENIOR", "50 ans et plus"
    # Exclusive to AI and ACI
    PARTICULAR_DIFFICULTIES = (
        "PARTICULAR_DIFFICULTIES",
        "Difficultés particulièrement importantes dont l'absence de prise en charge ferait "
        "obstacle à son insertion professionnelle",
    )
    # Since December 1, 2021, health context reason can no longer be used
    HEALTH_CONTEXT = "HEALTH_CONTEXT", "Contexte sanitaire"

    @classmethod
    def for_company(cls, company):
        enums = [
            cls.SENIOR_CDI,
            cls.COMPLETE_TRAINING,
            cls.RQTH,
            cls.SENIOR,
        ]
        if company.kind in [companies_enums.CompanyKind.AI, companies_enums.CompanyKind.ACI]:
            enums.append(cls.PARTICULAR_DIFFICULTIES)

        empty = [(None, cls.__empty__)] if hasattr(cls, "__empty__") else []
        return empty + [(enum.value, enum.label) for enum in enums]


class ProlongationRequestStatus(models.TextChoices):
    PENDING = "PENDING", "À traiter"
    GRANTED = "GRANTED", "Acceptée"
    DENIED = "DENIED", "Refusée"


class ProlongationRequestDenyReason(models.TextChoices):
    IAE = "IAE", "L’IAE ne correspond plus aux besoins / à la situation de la personne."
    SIAE = "SIAE", "La typologie de SIAE ne correspond plus aux besoins / à la situation de la personne."
    DURATION = "DURATION", "La durée de prolongation demandée n’est pas adaptée à la situation du candidat."


class ProlongationRequestDenyProposedAction(models.TextChoices):
    # TODO: Clarify the actions to improve the naming
    EXIT_IAE = (
        "EXIT_IAE",
        "Accompagnement à la recherche d’emploi hors IAE et mobilisation de l’offre "
        "de services disponible au sein de votre structure ou celle d’un partenaire.",
    )
    SOCIAL_PARTNER = "SOCIAL_PARTNER", "Orientation vers un partenaire de l’accompagnement social/professionnel."
    OTHER = "OTHER", "Autre"
