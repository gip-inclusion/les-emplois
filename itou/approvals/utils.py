from itou.companies.enums import SIAE_WITH_CONVENTION_KINDS


def get_user_last_accepted_siae_job_application(user):
    if not user.is_job_seeker:
        return None

    # Some candidates may not have accepted job applications
    # Assuming it's the case can lead to issues downstream
    return (
        user.job_applications.accepted()
        .filter(to_company__kind__in=SIAE_WITH_CONVENTION_KINDS)
        .with_accepted_at()
        .order_by("-accepted_at", "-hiring_start_at")
        .first()
    )


def last_hire_was_made_by_siae(user, siae):
    if not user.is_job_seeker:
        return False
    last_accepted_job_application = get_user_last_accepted_siae_job_application(user)
    return last_accepted_job_application and last_accepted_job_application.to_company_id == siae.pk
