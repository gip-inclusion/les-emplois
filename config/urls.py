from django.conf import settings
from django.contrib import admin
from django.urls import include, path, re_path, register_converter
from django.views.generic import TemplateView

from itou.utils.urls import SiretConverter
from itou.www.dashboard import views as dashboard_views
from itou.www.login import views as login_views
from itou.www.signup import views as signup_views


register_converter(SiretConverter, "siret")

urlpatterns = [
    path("admin/", admin.site.urls),
    # --------------------------------------------------------------------------------------
    # allauth URLs. Order is important because some URLs are overriden.
    # --------------------------------------------------------------------------------------
    # Override allauth `account_signup` URL.
    # /accounts/signup/ <=> account_signup
    # We don't want any user to be able to signup using the default allauth `signup` url
    # because we have multiple specific signup processes for different kind of users.
    re_path(r"^accounts/signup/$", signup_views.signup),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_login` URL.
    # /accounts/login/ <=> account_login
    # We override this view because the login page should look slightly differently
    # for job seekers, prescribers and employers.
    # Also, PEAMU is only available for job seekers.
    re_path(r"^accounts/login/$", login_views.login),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_change_password` URL.
    # /accounts/password/change/ <=> account_change_password
    # https://github.com/pennersr/django-allauth/issues/468
    re_path(r"^accounts/password/change/$", dashboard_views.password_change),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_reset_password` URL.
    # Avoid user enumeration via password reset page.
    re_path(r"^accounts/password/reset/$", signup_views.ItouPasswordResetView.as_view()),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_logout` URL.
    # /accounts/logout/ <=> account_logout
    # We need custom code to process PEAMU logout.
    re_path(r"^accounts/logout/$", dashboard_views.logout),
    # --------------------------------------------------------------------------------------
    # Other allauth URLs.
    path("accounts/", include("allauth.urls")),
    # --------------------------------------------------------------------------------------
    # PEAMU URLs.
    path("accounts/", include("itou.allauth_adapters.peamu.urls")),
    # FranceConnect URLs.
    path("franceconnect/", include("itou.france_connect.urls")),
    # --------------------------------------------------------------------------------------
    # API.
    path("api/v1/", include("itou.api.urls", namespace="v1")),
    path("api/", include("itou.www.api.urls")),
    # www.
    path("", include("itou.www.home.urls")),
    path("apply/", include("itou.www.apply.urls")),
    path("approvals/", include("itou.www.approvals_views.urls")),
    path("autocomplete/", include("itou.www.autocomplete.urls")),
    path("dashboard/", include("itou.www.dashboard.urls")),
    path("employee_record/", include("itou.www.employee_record_views.urls")),
    path("institutions/", include("itou.www.institutions_views.urls")),
    path("invitations/", include("itou.www.invitations_views.urls")),
    path("prescribers/", include("itou.www.prescribers_views.urls")),
    path("search/", include("itou.www.search.urls")),
    path("siae/", include("itou.www.siaes_views.urls")),
    path("signup/", include("itou.www.signup.urls")),
    path("stats/", include("itou.www.stats.urls")),
    path("versions/", include("itou.www.releases.urls")),
    # Enable Anymailâ€™s status tracking
    # https://anymail.readthedocs.io/en/stable/esps/mailjet/#status-tracking-webhooks
    re_path(r"^webhooks/anymail/", include("anymail.urls")),
    path("welcoming_tour/", include("itou.www.welcoming_tour.urls")),
    # Static pages.
    path("accessibility/", TemplateView.as_view(template_name="static/accessibility.html"), name="accessibility"),
    # Errors pages.
    path("403/", TemplateView.as_view(template_name="403.html"), name="403"),
    path("404/", TemplateView.as_view(template_name="404.html"), name="404"),
    path("500/", TemplateView.as_view(template_name="500.html"), name="500"),
]

if settings.DEBUG and "debug_toolbar" in settings.INSTALLED_APPS:
    import debug_toolbar

    urlpatterns = [path("__debug__/", include(debug_toolbar.urls))] + urlpatterns
    # In the local developpemnt environment, we enable the default callback URLs
    urlpatterns = [path("", include("itou.france_connect.urls"))] + urlpatterns
