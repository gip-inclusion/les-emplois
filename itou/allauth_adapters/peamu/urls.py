from django.urls import include, re_path

from itou.allauth_adapters.peamu import views as peamu_views
from itou.allauth_adapters.peamu.provider import PEAMUProvider


def default_urlpatterns(provider):
    urlpatterns = [
        re_path(r"^login/$", peamu_views.oauth2_login, name=provider.id + "_login"),
        re_path(r"^login/callback/$", peamu_views.oauth2_callback, name=provider.id + "_callback"),
    ]

    return [
        re_path(r"^profile/$", peamu_views.redirect_to_dashboard_view),
        re_path(r"^" + provider.get_slug() + r"/", include(urlpatterns)),
    ]


urlpatterns = default_urlpatterns(PEAMUProvider)
