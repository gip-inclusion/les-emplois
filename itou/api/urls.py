from django.urls import include, path
from drf_spectacular.views import SpectacularAPIView, SpectacularRedocView
from rest_framework import routers
from rest_framework.authtoken import views as auth_views

from itou.api.data_inclusion_api.views import DataInclusionStructureView

from .applicants_api.views import ApplicantsView
from .employee_record_api.viewsets import (
    DummyEmployeeRecordViewSet,
    EmployeeRecordUpdateNotificationViewSet,
    EmployeeRecordViewSet,
)
from .siae_api.viewsets import SiaeViewSet


# High level app for API
# https://docs.djangoproject.com/en/dev/topics/http/urls/#url-namespaces-and-included-urlconfs
app_name = "itou.api"

# Using DRF router with viewsets means automatic definition of utl patterns
router = routers.DefaultRouter()
router.register(r"employee-records", EmployeeRecordViewSet, basename="employee-records")
router.register(
    r"employee-record-notifications",
    EmployeeRecordUpdateNotificationViewSet,
    basename="employee-record-notifications",
)
router.register(r"dummy-employee-records", DummyEmployeeRecordViewSet, basename="dummy-employee-records")
router.register(r"siaes", SiaeViewSet, basename="siaes")

urlpatterns = [
    # TokenAuthentication endpoint to get token from login/password.
    path("token-auth/", auth_views.obtain_auth_token, name="token-auth"),
    # Needed for Browseable API (dev)
    path("api-auth/", include("rest_framework.urls", namespace="rest_framework")),
    # OpenAPI
    # See: https://www.django-rest-framework.org/topics/documenting-your-api/
    # OAS 3 YAML schema (downloadable)
    path(
        "oas3/",
        SpectacularAPIView.as_view(),
        name="openapi_schema",
    ),
    path(
        "redoc/",
        SpectacularRedocView.as_view(url_name="v1:openapi_schema"),
        name="redoc",
    ),
]

urlpatterns += router.urls

urlpatterns += [
    path("structures/", DataInclusionStructureView.as_view(), name="structures-list"),
    path("candidats/", ApplicantsView.as_view(), name="applicants-list"),
]
