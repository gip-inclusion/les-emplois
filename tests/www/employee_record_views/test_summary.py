import pytest
from django.urls import reverse
from freezegun import freeze_time
from pytest_django.asserts import assertContains, assertNotContains

from itou.employee_record.enums import Status
from itou.utils.templatetags.format_filters import format_approval_number, format_siret
from tests.companies.factories import CompanyWithMembershipAndJobsFactory
from tests.employee_record.factories import EmployeeRecordUpdateNotificationFactory, EmployeeRecordWithProfileFactory
from tests.job_applications.factories import JobApplicationWithCompleteJobSeekerProfileFactory
from tests.utils.test import parse_response_to_soup


class TestSummaryEmployeeRecords:
    @pytest.fixture(autouse=True)
    def setup_method(self):
        # User must be super user for UI first part (tmp)
        self.company = CompanyWithMembershipAndJobsFactory(name="Wanna Corp.", membership__user__first_name="Billy")
        self.user = self.company.members.get(first_name="Billy")
        self.job_application = JobApplicationWithCompleteJobSeekerProfileFactory(
            to_company=self.company, job_seeker__first_name="Lauren", job_seeker__last_name="Mata"
        )
        self.employee_record = EmployeeRecordWithProfileFactory(job_application=self.job_application)
        self.url = reverse("employee_record_views:summary", args=(self.employee_record.id,))

    def test_access_granted(self, client):
        # Must have access
        client.force_login(self.user)
        response = client.get(self.url)
        assert response.status_code == 200

    def test_check_job_seeker_infos(self, client):
        # Must have access
        client.force_login(self.user)
        response = client.get(self.url)
        profile = self.job_application.job_seeker.jobseeker_profile
        assertContains(
            response, f"<li>À : {profile.birth_place} ({profile.birth_place.department_code})</li>", count=1
        )

    def test_asp_batch_file_infos(self, client):
        HORODATAGE = "Horodatage ASP"
        client.force_login(self.user)
        response = client.get(self.url)
        assertNotContains(response, HORODATAGE)

        self.employee_record.ready()
        self.employee_record.wait_for_asp_response(file="RIAE_FS_20210410130000.json", line_number=1, archive=None)

        response = client.get(self.url)
        assertContains(response, HORODATAGE)
        assertContains(response, "Création : <b>RIAE_FS_20210410130000")

        EmployeeRecordUpdateNotificationFactory(
            employee_record=self.employee_record, asp_batch_file="RIAE_FS_20210510130000.json"
        )
        response = client.get(self.url)
        assertContains(response, HORODATAGE)
        assertContains(response, "Création : <b>RIAE_FS_20210410130000")
        assertContains(response, "Modification : <b>RIAE_FS_20210510130000")

    def test_technical_infos(self, client):
        client.force_login(self.user)
        response = client.get(self.url)

        assertContains(response, format_approval_number(self.employee_record.approval_number))
        assertContains(response, format_siret(self.employee_record.siret))
        assertContains(response, self.employee_record.asp_measure)

    @freeze_time("2025-04-29 11:11:11")
    @pytest.mark.parametrize(
        "status",
        [
            Status.NEW,
            Status.READY,
            Status.SENT,
            Status.REJECTED,
            Status.DISABLED,
            Status.ARCHIVED,
            Status.PROCESSED,
        ],
    )
    def test_action_bar(self, client, status, snapshot):
        self.employee_record.status = status
        self.employee_record.save()

        client.force_login(self.user)
        response = client.get(self.url)
        title_section_soup = parse_response_to_soup(
            response,
            selector=".s-title-02__col",
            replace_in_attr=[
                (
                    "href",
                    f"/employee_record/reactivate/{self.employee_record.pk}",
                    "/employee_record/reactivate/[Pk of EmployeeRecord]",
                ),
                (
                    "href",
                    f"/employee_record/create/{self.employee_record.job_application_id}",
                    "/employee_record/create/[Pk of JobApplication]",
                ),
                (
                    "href",
                    f"/employee_record/disable/{self.employee_record.pk}",
                    "/employee_record/disable/[Pk of EmployeeRecord]",
                ),
            ],
        )

        assert str(title_section_soup) == snapshot
