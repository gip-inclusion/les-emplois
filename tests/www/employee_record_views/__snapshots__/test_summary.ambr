# serializer version: 1
# name: TestSummaryEmployeeRecords.test_action_bar[ARCHIVED]
  '''
  <div class="s-title-02__col col-12">
                              
      
  
  
      <div class="c-prevstep">
          <a aria-label="Retour à la liste" class="btn btn-ico btn-link ps-0" href="/employee_record/list">
              <i aria-hidden="true" class="ri-arrow-drop-left-line ri-xl fw-medium"></i>
              <span>
                  Retour
                  à la liste
              </span>
          </a>
      </div>
  
  
  
                              
      
  <div class="c-title">
      <div class="c-title__main">
              <h1>
                  Fiche salarié ASP : Lauren MATA
                  
      <span class="badge rounded-pill badge-base text-nowrap bg-emploi-lightest text-primary">
          Archivée le 29/04/2025 13:11
      </span>
  
  
              </h1>
          </div>
      
      
  </div>
  
  
                              
                                  
  
  
  
                              
                              
      
  
                          </div>
  '''
# ---
# name: TestSummaryEmployeeRecords.test_action_bar[DISABLED]
  '''
  <div class="s-title-02__col col-12">
                              
      
  
  
      <div class="c-prevstep">
          <a aria-label="Retour à la liste" class="btn btn-ico btn-link ps-0" href="/employee_record/list">
              <i aria-hidden="true" class="ri-arrow-drop-left-line ri-xl fw-medium"></i>
              <span>
                  Retour
                  à la liste
              </span>
          </a>
      </div>
  
  
  
                              
      
  <div class="c-title">
      <div class="c-title__main">
              <h1>
                  Fiche salarié ASP : Lauren MATA
                  
      <span class="badge rounded-pill badge-base text-nowrap bg-emploi-lightest text-primary">
          Désactivée le 29/04/2025 13:11
      </span>
  
  
              </h1>
          </div>
      
      
  </div>
  
  
                              
                                  
  
  
  
                              
                              
      
          <div class="c-box c-box--action">
              <div class="form-row align-items-center gx-3">
                  
                      <div class="form-group col-12 col-lg-auto">
                          <a class="btn btn-lg btn-white btn-block btn-ico" href="/employee_record/reactivate/[Pk of EmployeeRecord]">
                              <i aria-hidden="true" class="ri-reset-right-line font-weight-medium"></i>
                              <span>Réactiver</span>
                          </a>
                      </div>
                  
                  
              </div>
          </div>
      
  
                          </div>
  '''
# ---
# name: TestSummaryEmployeeRecords.test_action_bar[NEW]
  '''
  <div class="s-title-02__col col-12">
                              
      
  
  
      <div class="c-prevstep">
          <a aria-label="Retour à la liste" class="btn btn-ico btn-link ps-0" href="/employee_record/list">
              <i aria-hidden="true" class="ri-arrow-drop-left-line ri-xl fw-medium"></i>
              <span>
                  Retour
                  à la liste
              </span>
          </a>
      </div>
  
  
  
                              
      
  <div class="c-title">
      <div class="c-title__main">
              <h1>
                  Fiche salarié ASP : Lauren MATA
                  
      <span class="badge rounded-pill badge-base text-nowrap bg-emploi">Nouvelle à compléter</span>
  
  
              </h1>
          </div>
      
      
  </div>
  
  
                              
                                  
  
  
  
                              
                              
      
          <div class="c-box c-box--action">
              <div class="form-row align-items-center gx-3">
                  
                      <div class="form-group col-12 col-lg-auto">
                          <a class="btn btn-lg btn-white btn-block btn-ico" href="/employee_record/create/[Pk of JobApplication]?from_status=NEW">
                              <i aria-hidden="true" class="ri-file-edit-line font-weight-medium"></i>
                              <span>Compléter</span>
                          </a>
                      </div>
                  
                  
              </div>
          </div>
      
  
                          </div>
  '''
# ---
# name: TestSummaryEmployeeRecords.test_action_bar[PROCESSED]
  '''
  <div class="s-title-02__col col-12">
                              
      
  
  
      <div class="c-prevstep">
          <a aria-label="Retour à la liste" class="btn btn-ico btn-link ps-0" href="/employee_record/list">
              <i aria-hidden="true" class="ri-arrow-drop-left-line ri-xl fw-medium"></i>
              <span>
                  Retour
                  à la liste
              </span>
          </a>
      </div>
  
  
  
                              
      
  <div class="c-title">
      <div class="c-title__main">
              <h1>
                  Fiche salarié ASP : Lauren MATA
                  
      <span class="badge rounded-pill badge-base text-nowrap bg-success">
          Intégrée par l'ASP le 29/04/2025 13:11
      </span>
  
  
              </h1>
          </div>
      
      
  </div>
  
  
                              
                                  
  
  
  
                              
                              
      
          <div class="c-box c-box--action">
              <div class="form-row align-items-center gx-3">
                  
                  
                      <div class="form-group col-12 col-lg-auto">
                          <a class="btn btn-lg btn-outline-white btn-block btn-ico" href="/employee_record/disable/[Pk of EmployeeRecord]">
                              <i aria-hidden="true" class="ri-file-close-line font-weight-medium"></i>
                              <span>Désactiver</span>
                          </a>
                      </div>
                  
              </div>
          </div>
      
  
                          </div>
  '''
# ---
# name: TestSummaryEmployeeRecords.test_action_bar[READY]
  '''
  <div class="s-title-02__col col-12">
                              
      
  
  
      <div class="c-prevstep">
          <a aria-label="Retour à la liste" class="btn btn-ico btn-link ps-0" href="/employee_record/list">
              <i aria-hidden="true" class="ri-arrow-drop-left-line ri-xl fw-medium"></i>
              <span>
                  Retour
                  à la liste
              </span>
          </a>
      </div>
  
  
  
                              
      
  <div class="c-title">
      <div class="c-title__main">
              <h1>
                  Fiche salarié ASP : Lauren MATA
                  
      <span class="badge rounded-pill badge-base text-nowrap bg-secondary">
          Complétée le 29/04/2025 13:11
      </span>
  
  
              </h1>
          </div>
      
      
  </div>
  
  
                              
                                  
  
  
  
                              
                              
      
  
                          </div>
  '''
# ---
# name: TestSummaryEmployeeRecords.test_action_bar[REJECTED]
  '''
  <div class="s-title-02__col col-12">
                              
      
  
  
      <div class="c-prevstep">
          <a aria-label="Retour à la liste" class="btn btn-ico btn-link ps-0" href="/employee_record/list">
              <i aria-hidden="true" class="ri-arrow-drop-left-line ri-xl fw-medium"></i>
              <span>
                  Retour
                  à la liste
              </span>
          </a>
      </div>
  
  
  
                              
      
  <div class="c-title">
      <div class="c-title__main">
              <h1>
                  Fiche salarié ASP : Lauren MATA
                  
      <span class="badge rounded-pill badge-base text-nowrap bg-danger">
          Retour en erreur le 29/04/2025 13:11
      </span>
  
  
              </h1>
          </div>
      
      
  </div>
  
  
                              
                                  
  
  
  
                              
                              
      
          <div class="c-box c-box--action">
              <div class="form-row align-items-center gx-3">
                  
                      <div class="form-group col-12 col-lg-auto">
                          <a class="btn btn-lg btn-white btn-block btn-ico" href="/employee_record/create/[Pk of JobApplication]?from_status=REJECTED">
                              <i aria-hidden="true" class="ri-pencil-line font-weight-medium"></i>
                              <span>Modifier</span>
                          </a>
                      </div>
                  
                  
                      <div class="form-group col-12 col-lg-auto">
                          <a class="btn btn-lg btn-outline-white btn-block btn-ico" href="/employee_record/disable/[Pk of EmployeeRecord]">
                              <i aria-hidden="true" class="ri-file-close-line font-weight-medium"></i>
                              <span>Désactiver</span>
                          </a>
                      </div>
                  
              </div>
          </div>
      
  
                          </div>
  '''
# ---
# name: TestSummaryEmployeeRecords.test_action_bar[SENT]
  '''
  <div class="s-title-02__col col-12">
                              
      
  
  
      <div class="c-prevstep">
          <a aria-label="Retour à la liste" class="btn btn-ico btn-link ps-0" href="/employee_record/list">
              <i aria-hidden="true" class="ri-arrow-drop-left-line ri-xl fw-medium"></i>
              <span>
                  Retour
                  à la liste
              </span>
          </a>
      </div>
  
  
  
                              
      
  <div class="c-title">
      <div class="c-title__main">
              <h1>
                  Fiche salarié ASP : Lauren MATA
                  
      <span class="badge rounded-pill badge-base text-nowrap bg-warning">
          Envoyée à l'ASP le 29/04/2025 13:11
      </span>
  
  
              </h1>
          </div>
      
      
  </div>
  
  
                              
                                  
  
  
  
                              
                              
      
  
                          </div>
  '''
# ---
